esp/esdllib/esdl_script.cpp

```diff
@@ -115,11 +176,22 @@ class CEsdlTransformOperationVariable : public CEsdlTransformOperationBase
     {
     }
 
-    virtual bool process(IEsdlScriptContext * context, IPropertyTree *content, IXpathContext * xpathContext) override
+    virtual bool process(IEsdlScriptContext * scriptContext, IXpathContext * targetContext, IXpathContext * sourceContext) override
     {
         if (m_select)
-            return xpathContext->addCompiledVariable(m_name, m_select);
-        return xpathContext->addVariable(m_name, "");
+            return sourceContext->addCompiledVariable(m_name, m_select);
+        else if (m_children.length())
```

Comment:
This case assigns the value of the variable to the result of processing the children?

Reply:
Yes, exactly.  It creates a temporary location for that xml in a special section of the script context tree and then changes the target for processing of the children to that location.  The end result is structured xml content stored in the variable.